torch_geometric/transforms/remove_isolated_nodes.py

```diff
@@ -11,18 +13,91 @@
 class RemoveIsolatedNodes(BaseTransform):
     r"""Removes isolated nodes from the graph
     (functional name: :obj:`remove_isolated_nodes`)."""
-    def __call__(self, data):
-        num_nodes = data.num_nodes
-        out = remove_isolated_nodes(data.edge_index, data.edge_attr, num_nodes)
-        data.edge_index, data.edge_attr, mask = out
+    def __call__(self, data: Union[Data,
+                                   HeteroData]) -> Union[Data, HeteroData]:
 
-        if hasattr(data, '__num_nodes__'):
-            data.num_nodes = int(mask.sum())
+        if isinstance(data, Data):
+            num_nodes = data.num_nodes
+            out = remove_isolated_nodes(data.edge_index, data.edge_attr,
+                                        num_nodes)
+            data.edge_index, data.edge_attr, mask = out
 
-        for key, item in data:
+            if hasattr(data, '__num_nodes__'):
+                data.num_nodes = int(mask.sum())
+
+            for key, item in data:
+                if bool(re.search('edge', key)):
+                    continue
+                if torch.is_tensor(item) and item.size(0) == num_nodes:
+                    data[key] = item[mask]
+
+            return data
+
+        elif isinstance(data, HeteroData):
+            return remove_hetero_isolated_nodes(data)
+
+        raise TypeError(f"`RemoveIsolatedNodes` invalid type: {type(data)}")
+
+
+def remove_hetero_isolated_nodes(data: HeteroData):
```

Comment:
This cannot go in 'torch_geometric/utils/isolated.py` because that class is imported by `torch_geometric/data/hetero_data.py`.

I thought about writing this taking a dictionary of edge_index, in the same way that `remove_isolated_nodes` works, I can adapt it to work like this if wanted, but I'd rather the `remove_isolated_nodes` were just depreciated in favour of one taking a `Data` object.

Any thoughts?

Reply:
Yeah, handling this with `data` only sounds like a better approach :)